### DISCUSSION TITLE
Unable to generate a oct JWK with "k" property.

### DISCUSSION THREAD

**Discussion by whaaaley** [panva/jose#83] [Q&A]
This code outputs the following...
```js
JWK.generateSync('oct', 256, {
  alg: 'HS256',
  use: 'sig',
  key_ops: ['sign', 'verify']
})
```
```js
{
  alg: 'HS256',
  use: 'sig',
  key_ops: [ 'sign', 'verify' ],
  kid: 'swswvFOKmCXNRyn4DCT6Ktb5Lz39dvmw2Br6C9YDES8',
  kty: 'oct'
}
```

I'm new to dealing with JWT and JWK at this level, so it's likely I'm missing something obvious. I can't seem to get `jose` to output `"k"` rather than `"kid"`. I didn't want to file this as a bug; I just don't know. I've been researching but I can't find any solution. However I did find many JWK generators online, all of which include `"k"` instead of `"kid"`.

Out of curiosity I saved the JWK output by `jose` and renamed `"kid"` to `"k"` and it signed and verified the JWT just fine.

Am I doing something wrong or is this not working as intended?

**Comment by panva** [comment]
console output is **intentionally** not giving out private properties by default. You can access the `k` property normally. You can also run `key.toJWK(true)` to output a JWK object with the private properties. Please see the docs.

`kid` is the key id, again, please read the docs.
